tests/math/unary/test_nondifferentiable.py

```diff
@@ -0,0 +1,25 @@
+from mygrad import argmin, argmax
+from mygrad.tensor_base import Tensor
+
+import hypothesis.strategies as st
+from numpy.testing import assert_equal
+import numpy as np
+from hypothesis import given
+
+import numpy as np
+
+@given(d=st.integers(min_value=3, max_value=10))
+def test_argmin(d):
+    dimensions = tuple(np.random.randint(1,6) for i in range(d)) 
+
+    a = np.random.randint(-100,100, size=dimensions)
```

Comment:
This would be good to do over other data types as well, but Ryan can chime in with testing

Reply:
Yes, you can use the following hypothesis strategy to test various dtypes:

```python
dtype_strat_numpy = st.sampled_from((np.int8, np.int16, np.int32, np.int64,
                                     np.float16, np.float32, np.float64))
```

Additionally, you should use `hypothesis` to draw the arrays, rather than use numpy to generate them randomly. You can also use mygrad's custom search strategy `valid_axes` - this will generate the appropriate positive and negative axis values, in addition to `None`

```python
from ...custom_strategies import valid_axes

@given(a=hnp.arrays(shape=hnp.array_shapes(max_side=4, max_dims=5),
                    dtype=dtype_strat_numpy ,
                    elements=st.floats(-1e-6., 1.)),
       data=st.data())
def test_transpose(a, data):
    axis = data.draw(valid_axes(ndim=a.ndim, single_axis_only=True), label="axis")
```

Using the `st.data()` search strategy allows you  draw axes based on the array `a` that was generated via the `@given` decorator.

This is far superior to randomly drawing an array as:
- hypothesis does not merely generate random numbers, it specifically generates elements to find edge cases that break code. It also simplifies the fail case whenever possible before reporting it.
- you cannot recreate or even see the random array that happened to break your unit test. Hypothesis will nicely report all of the objects that it generated if the test fails

Reply:
Whoops! Good catch. Thanks.
(edited in the earlier post to fix this)

Reply:
I updated the code. One problem with the code you wrote though: it generates Float regardless of what the sampled dtype is, which results in Hypothesis complaining about downsampling when it tries to convert the Float into, say, Int, when Int is the sampled dtype. I just left the `element` parameter as `None`.